### DISCUSSION TITLE
how can i test component with next-translate?

### DISCUSSION THREAD

**Discussion by Cleysonlb** [vinissimus/next-translate#413] [General] [answered]
hey, I'm in doubt how I could test a component that has translation, for example:

```
import React from 'react';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';
import { render, screen } from '@testing-library/react';
import Home from '../index';
import I18nProvider from 'next-translate/I18nProvider';

describe('<Home />', () => {
  it('should render Home', async () => {
    render(
      <I18nProvider lang="en">
        <Home />
      </I18nProvider>
    );

    expect(
      screen.getByText('translation:home.description')
    ).toBeInTheDocument();
  });
});

```

should i only test the existence of the key?
even after passing the test a warning is displayed
![image](https://user-images.githubusercontent.com/7593949/102524854-4b638a80-4078-11eb-9f45-d42d6108e496.png)

**Comment by aralroca** [comment, marked as answer]
You can disable the warning adding a custom `logger` on `I18nProvider`. You can pass the `i18n.json` config to `I18nProvider` as a prop, example:

https://github.com/vinissimus/next-translate/blob/1.0.0/__tests__/Trans.test.js#L8

**Reply by Cleysonlb** [reply]
Thanks @aralroca :+1:
